mlrun/runtimes/base.py

```diff
@@ -374,15 +374,16 @@ def _get_db_run(self, task: RunObject = None):
         if task:
             return task.to_dict()
 
-    def _generate_runtime_env(self, runobj: RunObject):
+    def generate_runtime_env(self, runobj: RunObject = None):
```

Comment:
add some docs and move function above as we (mostly try) keep public functions up, private functions down

Reply:
Good point. I created the public function you suggested on another comment and moved it up.